### ISSUE TITLE
lint/mixedRecv: warn when there is a mix of pointer no-pointer receivers

### ISSUE THREAD

**Issue by cristaloleg** [go-critic/go-critic#331]
Name: `mixedRecv`
Before:
```go
func (r Struct) MethA() {} // that's strange
func (r *Struct) MethB() {}
func (r *Struct) MethC() {}
```
After:
```go
func (r *Struct) MethA() {}  // OK
func (r *Struct) MethB() {}
func (r *Struct) MethC() {}
```

**Comment by cristaloleg** [comment]
LGTM

**Comment by Quasilyte** [comment]
Notice for future: may want to permit some cases.
Example for that: https://play.golang.org/p/YdcbyFkWD5y

**Comment by NexoMichael** [comment]
Maybe also check for different names:

Before:

```
func (r Struct) MethA() {} // that's strange
func (r2 Struct) MethB() {} // that's twice strange
func (r *Struct) MethC() {}
func (r *Struct) MethD() {}
func (newName *Struct) MethE() {} // name changed?

```
After:
```
func (r *Struct) MethA() {}  // OK
func (r *Struct) MethB() {} // OK
func (r *Struct) MethC() {}
func (r *Struct) MethD() {}
func (r *Struct) MethE() {} // OK
```

it is also need to be defined is it better to have "maximum wins" or "first wins" approach? seems like "maximum" wins is better with fallback to "first wins" in case if counts are equal

**Comment by Quasilyte** [comment]
> Maybe also check for different names:

`golint` does that alread. ;)
It reports inconsistent recv names.

**Comment by cristaloleg** [comment]
> it is also need to be defined is it better to have "maximum wins" or "first wins" approach? seems like "maximum" wins is better with fallback to "first wins" in case if counts are equal

From my experience having `(r Struct)` while there are `(r *Struct)` is a mistype. 
Anyway it's open for a discussion.

BTW we've an idea to exclude `Stringer`, sometimes it's needed.

**Comment by Antonboom** [comment]
Related:

1. https://www.jetbrains.com/help/go/code-inspections-in-go.html#Probable_bugs (Mixed value and pointer receivers)
2. https://github.com/dominikh/go-tools/issues/1337
